NaReguaApp/src/contexts/auth.js

```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useState, useEffect, useContext } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import api from '../services/api';
+
+const AuthContext = createContext({});
+
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
+	const [loading, setLoading] = useState(true);
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
+				setIsProvider(storageUser.isProvider);
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
+			}
+			setLoading(false);
+		}
+
+		carregarStorage();
+	}, []);
+
+	async function login(data) {
+		try {
+			setLoading(true);
+			let res = await api.post('/login', data);
+
+			let { token } = res.data;
```

Comment:
Codacy has a fix for the issue: [Replace `↹↹↹let·{·token·` with `······let·{token`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      let {token} = res.data;
```